CTX7-971: Client Metrics

fahreddinozcan opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added HTTP health check endpoint.
  * Enhanced client context/telemetry to include IDE and version, propagated through requests.
  * Improved HTTP session handling (randomized session IDs) and clientInfo caching for downstream calls.

* **Bug Fixes**
  * Better JSON-RPC 2.0 error responses and more robust contextual error handling.
  * Improved telemetry/cache maintenance (probabilistic cleanup) to reduce stale entries.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

fahreddinozcan committed:
feat: telemetry for client and tool call infos

fahreddinozcan committed:
remove telemetry lib

fahreddinozcan committed:
feat: simplify metric collection

fahreddinozcan committed:
fix: remove server version duplication

fahreddinozcan committed:
fix: revert session id handling

fahreddinozcan committed:
feat: simplify header generation for client metadata

fahreddinozcan committed:
feat: simplify client context and header params

fahreddinozcan committed:
feat: getClientContext method for easy access

fahreddinozcan committed:
Merge branch 'master' into CTX7-971-v2

Resolved conflicts by integrating 2.0.0 API changes with simplified context handling:
- Updated searchLibraries to use new query+libraryName params with ClientContext
- Updated fetchLibraryContext (new in 2.0.0) with ClientContext support
- Kept getClientContext() helper for simplified tool handler context passing
- All tool handlers now use getClientContext() for cleaner code

fahreddinozcan committed:
feat: finalize metric collection for stdio

enesgules commented on packages/mcp/src/index.ts:
```diff
@@ -4,12 +4,15 @@ import { McpServer } from "@modelcontextprotocol/sdk/server/mcp.js";
 import { Command } from "commander";
 import { AsyncLocalStorage } from "async_hooks";
 
+export const SERVER_VERSION = "2.0.0";
```
let's update the release action to modify this on release

enesgules committed:
Merge remote-tracking branch 'origin' into CTX7-971-v2

enesgules committed:
feat: move SERVER_VERSION to constants file and update version injection in workflows

Moved SERVER_VERSION constant from index.ts to lib/constants.ts for better organization. Updated GitHub workflows to inject version into the new location during ECR deployment and release processes.

fahreddinozcan committed:
Merge branch 'master' of https://github.com/upstash/context7 into CTX7-971-v2

enesgules committed:
feat: refactor global state management for stdio vs HTTP modes

fahreddinozcan committed:
feat: normalize client name

fahreddinozcan committed:
fix: remove client normalization

fahreddinozcan committed:
chore: add changeset
